src/index.ts

```diff
@@ -29,32 +29,45 @@ const gateURI = 'http://localhost:3000';
  *               to the webapp's backend
  *
  */
+
 // instantation, everything before the return callback runs only once
 export default function gatelog(projectID: string, apiKey: string) {
-    // const newAuth = new AuthVerification(gateURI, projectID, apiKey);
-
-    // const validate = newAuth.endpointValidation(req, res, next);
-    // const verify = newAuth.keyVerification(req, res, next);
-
-    // run the API Key verification process when gateLogger is instantiated
-    try {
-        // await validate();
-        // await verify();
-    } catch (err) {
-        return err;
-    }
-
-    // const postQuery = new PostQuery(gateURI, projectID);
-
-    // every time a request is processed in the user's backend
-    return async (req: Request, res: Response, next: NextFunction) => {
-        // reassign res.end in order to allow logger functionality before
-        // a response is sent back the client
-        // eslint-disable-next-line arrow-body-style
-        res.end = () => {
-            // postQuery.post(req, res, next);
-            return res;
-        };
+    const newAuth = new AuthVerification(gateURI, projectID, apiKey);
+
+    const validate = newAuth.validation;
+    const verify = newAuth.verification;
+
+    // run the API Key verification process when gatelog is instantiated
+    validate().catch((err) => err);
```

Comment:
These catches isn't really doing anything and the function will still return middleware regardless of whether the verification is successful. I think we should only return middleware if the API key verification is successful, if not we should throw an error letting the developer know the API key is invalid.

Reply:
changed it so that both of these functions either return true or an error, and if they do not return true, we return an error instead of returning the middleware callback